index.js

```diff
@@ -397,6 +410,46 @@ async function fastifyStatic (fastify, opts) {
     const routeConfig = req.routeOptions?.config || req.routeConfig
     pumpSendToReply(req, reply, routeConfig.file, routeConfig.rootPath)
   }
+
+  async function generateHashForFiles () {
+    fileHashes = new Map()
+
+    const roots = Array.isArray(sendOptions.root) ? sendOptions.root : [sendOptions.root]
+    for (let rootPath of roots) {
+      rootPath = rootPath.split(path.win32.sep).join(path.posix.sep)
+      if (!rootPath.endsWith('/')) {
+        rootPath += '/'
+      }
+
+      const files = await glob(`${rootPath}**/**`, { follow: true, nodir: true, dot: opts.serveDotFiles })
```

Comment:
if we are doing this on the startup and block loading anyway, wonder if it would make sense to use `globSync` instead and make whole function synchronous.

Reply:
This was actually the case but mid work I upgraded to glob v10 (merged to main) and forgot to modify